src/core/abr/representation_chooser.ts

```diff
@@ -327,170 +333,198 @@ export default class RepresentationChooser {
     bitrate: undefined|number; // bitrate estimation
     representation: Representation|null; // chosen representation
   }> {
-    if (representations.length < 2) {
-      return observableOf({
-        bitrate: undefined, // Bitrate estimation is deactivated here
-        representation: representations.length ?
-          representations[0] : null,
-      })
-        .pipe(takeUntil(this._dispose$));
-    }
-
-    const {
-      manualBitrate$,
-      maxAutoBitrate$,
-      _initialBitrate,
-    }  = this;
-
-    const _deviceEventsArray : Array<Observable<IFilters>> = [];
-
-    if (this._limitWidth$) {
-      _deviceEventsArray.push(
-        this._limitWidth$
-          .pipe(map(width => ({ width })))
+    const authorizedRepresentations$: Observable<Representation[]> =
+      this._authorizations$.pipe(
+        map((authorizations) => {
+          return authorizations
+            .filter((authorization) => {
+              const authId = authorization.IDSet.representationId;
+              const rep = representations.find((_rep) => _rep.id === authId);
+              if (rep) {
+                return authorization.isPlayable;
+              }
+              return false;
+            })
+            .map((auth) => {
+              return representations.find((rep) =>
+                rep.id === auth.IDSet.representationId);
+            });
+        }),
+        startWith(representations)
       );
-    }
 
-    if (this._throttle$) {
-      _deviceEventsArray.push(
-        this._throttle$
-          .pipe(map(bitrate => ({ bitrate })))
-      );
-    }
+    return authorizedRepresentations$.pipe(
+      switchMap((authorizedRepresentations) => {
+        if (authorizedRepresentations.length === 0) {
+          throw new Error("ABR: No representations can be played, \
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Use string concatenation rather than line continuation. ([typescript:S1516](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1516))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=263&issues=AWSpExUpl6vpliCesIp3&open=AWSpExUpl6vpliCesIp3)

[ISSUE_KEY]: AWSpExUpl6vpliCesIp3